src/Bridge/Doctrine/Orm/ItemDataProvider.php

```diff
@@ -154,24 +152,42 @@ private function getIdentifiers(string $resourceClass, $id)
      * @param ObjectManager $resourceManager
      * @param string        $resourceClass
      *
+     * @throws RuntimeException
+     *
      * @return \Doctrine\ORM\QueryBuilder
      */
     private function retrieveQueryBuilder(ObjectManager $resourceManager, string $resourceClass)
     {
         $repository = $resourceManager->getRepository($resourceClass);
 
-        if ($repository instanceof  EntityRepository) {
+        if ($repository instanceof EntityRepository) {
             return $repository->createQueryBuilder('o');
         }
 
         if ($resourceManager instanceof EntityManagerInterface) {
```

Comment:
Hmm I don't understand why `EntityManagerInterface` would take priority upon `EntityRepository` if it has a `createQueryBuilder` method.

```
        if ($repository instanceof EntityRepository || method_exists($repository, 'createQueryBuilder')) {
            return $repository->createQueryBuilder('o');
        }

        if ($resourceManager instanceof EntityManagerInterface || method_exists($resourceManager, 'createQueryBuilder')) {
            return $resourceManager->createQueryBuilder()
                ->select('o')
                ->from($resourceClass, 'o');
        }

        throw new RuntimeException();
```  

I'd add that I find it weird to test that the `$resourceManager` has the `createQueryBuilder`, especially if it's not implementing the `EntityManagerInterface`. IMO this will almost always be false, what's the edge case that I'm missing?

IMO there is no better way of returning a query builder with a select clause on the correct entity, and it's a fine way of doing it.